react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+
+class PasswordForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: '',
+      validated: false,
+      error: 'req'
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ value: event.target.value });
+
+    if (event.target.value.length === 0) {
+      this.setState({ validated: false });
+      this.setState({ error: 'req' });
+    }
+    if (event.target.value.length > 0) {
+      this.setState({ validated: false });
+      this.setState({ error: 'short' });
+    }
+    if (event.target.value.length >= 8) {
+      this.setState({ validated: true });
+      this.setState({ error: '' });
+    }
+  }
+
+  analyzeRegex() {
+    const value = this.state.value;
+    if (passwordRegex.test(value)) {
+      return '';
+    } else {
+      return 'Your password must contain a digit, a capital letter and a special character.';
+    }
+  }
+
+  getErrorMsg() {
+    const errorMsg = this.state.error;
+    if (errorMsg === 'req') {
+      return 'A password is required.';
+    }
+    if (errorMsg === 'short') {
+      return 'Your password is too short.';
+    }
+    if (errorMsg === '') {
+      return '';
+    }
```

Comment:
Rather than storing short codes in state, you can store the whole error message, that should eliminate the need for this method.